src/Disruptor/RingBuffer`1.cs

```diff
@@ -137,10 +138,7 @@ public static RingBuffer<T> Create(ProducerType producerType, Func<T> factory, i
         public T this[long sequence]
         {
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get
-            {
-                return Util.Read<T>(_entries, _bufferPadRef + (int)(sequence & _indexMask));
-            }
+            get => Util.Read<T>(_entries, _bufferPadRef + (unchecked((int)sequence) & _indexMask));
```

Comment:
This should probably be `uint`, with `int` I cannot easily prove in my head it's always correct.

Reply:
I don't think you can cast sequence to `int` as you will truncate and any values over `int.MaxValue` will all return the same value.  I think you can only cast the result of the `& _indexMask` to something.

Reply:
Try this in interactive:

```
(unchecked((int)((long)int.MaxValue + 123))) & 1023
vs
(unchecked(((long)int.MaxValue + 123))) & 1023
```

Reply:
By casting to int we basically leave only the least significant 32 bits. With `ulong/uint` it's all nice and dandy, but with signed ones my mental model is not so strong.

Reply:
The good thing is that intermediate cast to unsigned ones almost never decreases performance, but is more readable:
```
get => Util.Read<T>(_entries, _bufferPadRef + (int)(unchecked((uint)sequence) & _indexMask));
```
Since `_indexMask` is positive `Pow2-1` int, cast from uint result of `and` to int is noop.

Reply:
Ah it works because `int.MaxValue` / `uint.MaxValue` are powers of two

Reply:
Hm, they are not powers of two, but masks - `2^31 - 1` for int and `2^32 - 1` for uint, and casting to them is the same as applying such a mask. But for signed ones there are often some special cases or rules for wrapping around, so it's simpler just to cast to unsigned, do "easy" bit manipulation, and cast back.

Reply:
You're right, overflowing will just wrap around (to `int.MinValue` for `int` or 0 for `uint`) and the mask with positive values (`uint`) or negative ones (`int`) will give the same result.  I had a brain fart and was visualising the cast as a truncation.